xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/web/XWikiURLFactory.java

```diff
@@ -97,14 +100,99 @@ URL createURL(String spaces, String name, String action, String querystring, Str
     URL createExternalURL(String spaces, String name, String action, String querystring, String anchor,
         String xwikidb, XWikiContext context);
 
+    /**
+     * Create a skin URL for the given filename and skin directory.
+     * @param filename the file to reach.
+     * @param skin the skin where the file should be loaded
+     * @param context current context
+     * @return a URL to load the given file.
+     */
     URL createSkinURL(String filename, String skin, XWikiContext context);
 
+    /**
+     * Create an URL for the given filename in the given skin directory.
+     * @param filename the file to reach.
+     * @param skin the skin where the file should be loaded
+     * @param context current context
+     * @param queryParameters the parameters to put at the end of the URL
+     * @return a URL to load the given file with the given query parameters.
+     * @since 11.1RC1
+     */
+    @Unstable
+    URL createSkinURL(String filename, String skin, XWikiContext context, Map<String, String> queryParameters);
```

Comment:
I just realized that Map<String, String> is actually too restrictive because it make impossible to have a parameter with several values. You could change for Map<String, Object> for example and check in the implementation if it's a Iterable or a table for example.

Reply:
Do you have a usecase in mind? I was thinking that in such cases, the user should create his own string from the table/list values, before submitting the query parameter.

Reply:
Hmmm. Right. And since I'm using Apache commons to create the URL, I should be able to rely on it for processing the parameters. So it should be easy.